File: CCC/CCC 18/CCC 18 J3 - Are we there yet/Are_we_there_yet.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){
    vector<int> p(5,0);
    for (int i=1; i<5; i++){
        cin >> p[i];
        p[i] += p[i-1];
    }

    for (int i=0; i<5; i++){
        for (int j=0; j<5; j++){
            cout << abs(p[i]-p[j]) << ' ';
        }
        cout << "\n";
    }

    return 0;
}

File: CCC/CCC 18/CCC 18 J3 - Are we there yet/Are_we_there_yet.py
import sys

d = map(int, sys.stdin.readline().strip().split())


def solve(d):
    p = [0]
    for j in d:
        p.append(p[-1] + j)

    for i in range(5):
        for j in range(5):
            ans = abs(p[i]-p[j])
            print(ans, end=' ')
        print()


ans_l = solve(d)

